### PR TITLE
add slr connectivity test

### PR DESCRIPTION
**Pull request by bobz965**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
- Tests

### <samp>🤖 Generated by Copilot at b229d8a</samp>

This pull request improves the logging, testing, and functionality of the load balancer features in `kube-ovn`. It adds more detailed messages to show the LB rule changes in `pkg/controller/endpoint.go`, updates the switch load balancer rule test cases in `test/e2e/kube-ovn/switch_lb_rule/switch_lb_rule.go`, enables the session affinity option for the TCP session load balancer in `pkg/controller/vpc.go`, and removes unused code in `pkg/controller/service.go`.

### <samp>🤖 Generated by Copilot at b229d8a</samp>

> _Sing, O Muse, of the code that the heroes refined_
> _To control the swift service and the endpoint divine_
> _They removed the old traces and added new logs_
> _And enabled the option to bind the TCP dogs_

### <samp>🤖 Generated by Copilot at b229d8a</samp>

*  Enable session affinity option for TCP session load balancer in VPC by modifying the `addLoadBalancer` function ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-ae8a8b5d44d277a4bb96635ee325fe54b48e8bc10b24a79d4bb8e211b57131ecL214-R214))
*  Add logging statements to provide more information about the update and deletion of LB rules in verbose mode in the `handleUpdateEndpoint` function ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528R197), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-e446ff04750bbbe87ec6cf1387960a29d56757d84784b18f2814a6e0be5fb528L202-R208))
*  Remove unused code and imports from the `enqueueDeleteService` function ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-b31c03e05358c6faab6e77c96909e14e14eeb0dc09412ede3329625adc197fa9L59), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L11))
*  Add and modify variables for the switch load balancer rule test cases in the `switch_lb_rule` package ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L30-R55), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L47-R63), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13R70-R72))
*  Change the pod image and command to run a netcat server on the `backendPort` for the switch load balancer rule test cases in the `switch_lb_rule` package ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L67-R84))
*  Change the port name and use variables instead of hard-coded values for the switch load balancer rule creation in the `switch_lb_rule` package ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L100-R121), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13L208-R233))
*  Add netcat calls to test the connectivity to the switch load balancer rule with session affinity enabled and disabled in the `switch_lb_rule` package ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13R199-R203), [link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13R310-R314))
*  Add an import statement for the `errors` package in the `switch_lb_rule` package ([link](https://github.com/kubeovn/kube-ovn/pull/3041/files?diff=unified&w=0#diff-0bc2dad9f41dd1a8d981f6e3b93597cb758775caa3758897ca73f7d2363e7f13R5))

### CODE REVIEW

File: test/e2e/framework/image.go [hunk 1]
```diff
@@ -4,4 +4,6 @@ const (
 	PauseImage   = "kubeovn/pause:3.2"
 	BusyBoxImage = "busybox:stable"
 	AgnhostImage = "kubeovn/agnhost:2.43"
+
+	BaseImageTemp = "kubeovn/kube-ovn-base:%s"
```
Comment (zhangzujian) [inline test/e2e/framework/image.go:8]: 不建议直接使用 base 镜像。如果需要测试 TCP，建议使用 agnhost 镜像对 HTTP 进行测试。

File: test/e2e/kube-ovn/switch_lb_rule/switch_lb_rule.go [hunk 2]
```diff
@@ -2,165 +2,278 @@ package switch_lb_rule
...
+			framework.Logf("output:\n%s", stdOutput)
+			framework.Logf("errOutput:\n%s", errOutput)
+			if i == 2 {
+				framework.Logf("exec %s failed err: %v, errOutput: %s, stdOutput: %s, retried %d times.", cmd, err, errOutput, stdOutput, i)
+				framework.ExpectNoError(errors.New("tcp netcat to svc %s failed"))
+			}
+			time.Sleep(6 * time.Second)
+		}
+	}
+	return ""
```
Comment (zhangzujian) [inline test/e2e/kube-ovn/switch_lb_rule/switch_lb_rule.go:57]: 是否可以使用 `framework.WaitUntil()` 替换掉这个循环？

File: test/e2e/kube-ovn/switch_lb_rule/switch_lb_rule.go [hunk 3]
```diff
@@ -2,165 +2,278 @@ package switch_lb_rule
...
 
-		ginkgo.By("Get pod " + podName)
-		pod, err = podClient.Get(context.TODO(), podName, metav1.GetOptions{})
+		ginkgo.By("Get client pod " + clientPodName)
+		clientPod, err = podClient.Get(context.TODO(), clientPodName, metav1.GetOptions{})
 		framework.ExpectNil(err)
-		framework.ExpectNotNil(pod)
+		framework.ExpectNotNil(clientPod)
+
+		stsVip := stsSvc.Spec.ClusterIPs[0]
```
Comment (zhangzujian) [inline test/e2e/kube-ovn/switch_lb_rule/switch_lb_rule.go:209]: 双栈集群中 ipv4 以及 ipv6 的 ClusterIP 都要检查。

File: pkg/controller/vpc.go [hunk 4]
```diff
@@ -211,7 +211,7 @@ func (c *Controller) addLoadBalancer(vpc string) (*VpcLoadBalancer, error) {
 	if err := c.initLB(vpcLbConfig.TcpLoadBalancer, string(v1.ProtocolTCP), false); err != nil {
 		return nil, err
 	}
-	if err := c.initLB(vpcLbConfig.TcpSessLoadBalancer, string(v1.ProtocolTCP), false); err != nil {
+	if err := c.initLB(vpcLbConfig.TcpSessLoadBalancer, string(v1.ProtocolTCP), true); err != nil {
```
Comment (zhangzujian) [inline pkg/controller/vpc.go:214]: 这是 bug 修复，单独提个 PR 吧。

**Comment by bobz965** [comment]
测试依赖指定ip来测试slr，但是指定 workload ip pool存在一个bug：
ovn.kubernetes.io/ip_pool:  设计上既可以为ip地址的集合，也可以为ip pool，一个key应该保持简单，不应有两种含义，我这边尝试添加一个的新的key进行解决。

**Comment by zhangzujian** [comment]
> 测试依赖指定ip来测试slr，但是指定 workload ip pool存在一个bug： ovn.kubernetes.io/ip_pool: 设计上既可以为ip地址的集合，也可以为ip pool，一个key应该保持简单，不应有两种含义，我这边尝试添加一个的新的key进行解决。

ippool 的 bug，与一个key是否进行了功能复用没有关系。

**Comment by bobz965** [comment]
> > 测试依赖指定ip来测试slr，但是指定 workload ip pool存在一个bug： ovn.kubernetes.io/ip_pool: 设计上既可以为ip地址的集合，也可以为ip pool，一个key应该保持简单，不应有两种含义，我这边尝试添加一个的新的key进行解决。
> 
> ippool 的 bug，与一个key是否进行了功能复用没有关系。

ok, 我后续重新梳力提交下 e2e

**Pull request closed without merging**